src/main/java/org/sert2521/gamename/Log.kt

```diff
@@ -0,0 +1,34 @@
+package org.sert2521.gamename
+
+import edu.wpi.first.wpilibj.Preferences
+
+/** Checks if a log is loggable with the provided log level from SmartDashboard. */
+private fun isLoggable(level: LogLevel) =
+        level >= when (Preferences.getInstance().getString("log_level", "info")) {
+            "verbose" -> LogLevel.VERBOSE
+            "debug" -> LogLevel.DEBUG
+            "warn" -> LogLevel.WARN
+            "error" -> LogLevel.ERROR
+            else -> LogLevel.INFO
+        }
+
+enum class LogLevel { VERBOSE, DEBUG, INFO, WARN, ERROR }
+
+private val autoTag: String get() {
+    Thread.currentThread().stackTrace
+            .filter { it.fileName != "LogKt" }
+            .forEach { return it.className }
+    return ""
+}
+
+fun verbose(tag: String? = null, message: Any) = log(LogLevel.VERBOSE, tag, message)
+fun debug(tag: String? = null, message: Any) = log(LogLevel.DEBUG, tag, message)
+fun info(tag: String? = null, message: Any) = log(LogLevel.INFO, tag, message)
+fun warn(tag: String? = null, message: Any) = log(LogLevel.WARN, tag, message)
+fun error(tag: String? = null, message: Any) = log(LogLevel.ERROR, tag, message)
```

Comment:
Nevermind, looks like it's not possible 🙁  (https://stackoverflow.com/a/34862984/2396125). Moving the tag to the second parameter wouldn't be _that_ weird, and might clean it up a bit. How do they do it at [Anko](https://github.com/Kotlin/anko/wiki/Anko-Commons-%E2%80%93-Logging#trait-like-style) that you linked above? I don't quite understand this or know if it's what we're looking for:

> [...] The methods require you to pass a tag argument. You can eliminate this with using AnkoLogger trait-like interface.